kubernetes-client-openapi/src/main/java/io/micronaut/kubernetes/client/openapi/ssl/DefaultPrivateKeyLoader.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.kubernetes.client.openapi.ssl;
+
+import io.micronaut.context.annotation.BootstrapContextCompatible;
+import jakarta.inject.Singleton;
+import org.bouncycastle.openssl.PEMKeyPair;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.math.BigInteger;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.RSAPrivateCrtKeySpec;
+import java.util.Base64;
+
+/**
+ * Default implementation of the private key loader which contains code copied from
+ * <a href="https://github.com/kubernetes-client/java/blob/v21.0.1/util/src/main/java/io/kubernetes/client/util/SSLUtils.java">SSLUtils</a>.
+ */
+@Singleton
+@BootstrapContextCompatible
+class DefaultPrivateKeyLoader implements KubernetesPrivateKeyLoader {
```

Comment:
@yawkat - this code has been copied from kubernetes-client (the link from javadoc). Please review and let me know if I need to change something